src/playwright/fixtures/mount.ts

```diff
@@ -1,52 +1,100 @@
-
-        const renderResult = render(template);
-        const html = await collectResult(renderResult);
-
         let pageError: Error | undefined;
 
         page.on('pageerror', (error: Error) => {
           pageError = error;
         });
 
-        const componentTags = await page.evaluate((html) => {
-          document.body.innerHTML = html;
+        await page.goto('/playwright.html');
 
+        // Now we leave Node.js Land and enter Browser Land:
+        //
+        // 1. Serialize both the template and its context so they make it across the wire,
+        //    and can be evaluated in the browser.
+        //
+        // 2. Evaluate and call the template function with its context.
+        //
+        // 3. Call Lit's `render()`, passing in the result of the evaluation and call.
+        await page.evaluate(
+          async ({ context, template }) => {
+            const { render } = await import('lit');
+
+            render(
+              // eslint-disable-next-line @typescript-eslint/no-implied-eval, @typescript-eslint/no-unsafe-call
+              await new Function(
+                'context',
+
+                // `html` is always referenced in templates, and `styleMap` sometimes is. So both
+                // need to be brought into scope.
+                //
+                // It's unlikely, but possible, that tests will use other Lit directives. If they
+                // do, those directives will also need to be brought into scope. When doing so,
+                // don't forget to update the import map in `playwright.html`.
+                //
+                // This approach simplifies tests. But you can imagine how it may not be suitable
+                // for a larger repository with more variation between tests, or in this repository
+                // if it sees more variation.
+                //
+                // An alternative approach would be to only evaluate the template and let tests
+                // bring themselves whatever they need into scope:
+                //
+                //   test('button', async ({ mount }) => {
+                //     await mount(async () => {
+                //       const { html } = await import('lit');
+                //
+                //       return html`<glide-core-button label="Label"></glide-core-button>`;
+                //     });
+                //    })
+                `
+                  return (async () => {
+                    const { html } = await import('lit');
+                    const { styleMap } = await import('lit/directives/style-map.js');
+
+                    return (${template})(eval('(' + context + ')'));
+                  })();
+                `,
+              )(context),
+              document.body,
+            );
+          },
+          {
+            context: serialize(context),
+            template: serialize(template),
+          },
+        );
+
+        const componentTags = await page.evaluate(() => {
```

Comment:
Put getting the tags in its own evaluation to clarify the purpose of this evaluation and the one above it that does the actual rendering. `page.evaluate()` calls aren't free. But they're also not restrictively expensive.